src/Simplex/Messaging/Server.hs

```diff
@@ -1957,21 +1974,6 @@ client
                     VRVerified q -> Right (q, t'')
                     VRFailed e -> Left (corrId', entId', ERR e)
 
-        deliverMessage :: T.Text -> QueueRec -> RecipientId -> Sub -> Maybe Message -> IO (Transmission BrokerMsg)
```

Comment:
this function unnecessarily duplicates the check for ProhibitSub at call sites and prevents different usage of messages in SUB and in ACK commands.